integration_tests/test_benchmark.py

```diff
@@ -18,22 +18,50 @@ def test_bench():
     os.chdir(get_repo_root_path())
 
     # Get numbers for current HEAD.
-    _run_cargo_bench("after")
+    return_code, stdout, stderr = _run_cargo_bench("after")
+    # Even if it is the first time this test is run, the benchmark tests should pass.
+    # For this purpose, we need to explicitly check the return code.
+    if return_code != 0:
+        print(stdout)
+        print(stderr)
+        assert return_code == 0
 
-    # Move to upstream tip.
-    subprocess.run(
-        "git fetch {} {}".format(REMOTE, BRANCH),
-        shell=True, check=True
-    )
-    subprocess.run(
-        "git checkout {}/{}".format(REMOTE, BRANCH),
-        shell=True, check=True
+    # Get numbers from upstream tip, without the changes from the current PR.
+    _checkout_master_branch()
+    return_code, stdout, stderr = _run_cargo_bench("before")
+
+    # Before checking any results, let's just go back to the PR branch.
+    # This way we make sure that the cleanup always happens even if the test fails.
+    _checkout_pr_branch()
+
+    if return_code == 0:
+        # In case this benchmark also ran successfully, we can call critcmp and compare the results.
+        _run_critcmp()
+    else:
+        # The benchmark did not run successfully, but it might be that it is because a benchmark does not exist.
+        # In this case, we do not want to fail the test.
+        if "error: no bench target named `main`" in stderr:
+            # This is a bit of a &*%^ way of checking if the benchmark does not exist.
+            # Hopefully it will be possible to check it in another way...soon
+            print("There are no benchmarks in master. No comparison can happen.")
+        else:
+            print(stdout)
+            print(stderr)
+            assert return_code == 0
+
+
+def _run_cargo_bench(baseline):
+    """Runs `cargo bench` and tags the baseline."""
+    process = subprocess.run(
```

Comment:
nit maybe rename `process` with `p` (same as in `_run_critcmp()`) or `proc`, this way the return line would be somehow shorter.